packages/cli/src/types/CliConfig.ts

```diff
@@ -0,0 +1,47 @@
+import { ColumnDefProperties } from '@synthql/queries';
+export interface CliConfig {
+    /**
+     * Internal. DO NOT USE/SET this option
+     * Specifies which draft of the JSON Schema standard the schema adheres to
+     */
+    $schema?: string;
+    /**
+     * The output directory for the generated files
+     * e.g. `'src/generated'`
+     */
+    out: string;
+    /**
+     * The default schema to use e.g. `'public'`
+     * This is similar to the `search_path` in PostgreSQL
+     */
+    defaultSchema: string;
+    /**
+     * The schemas to include in generation e.g. `['public']`
+     */
+    schemas: Array<string>;
+    /**
+     * The tables to include in generation e.g. `['users', 'accounts']`
+     */
+    tables?: Array<string>;
+    /**
+     * The table definitions for any tables you want
+     * to be overriden during the schema generation
+     */
+    schemaDefOverrides?: SchemaDefOverrides;
+}
+export interface SchemaDefOverrides {
+    /**
+     * The column definitions for each table to be overriden
+     * The key for each table should be the qualified name for the table,
+     * in the form `${schemaName.tableName}` e.g `'public.users'`
+     */
+    [x: string]: TableDefOverrides | undefined;
+}
+export interface TableDefOverrides {
```

Comment:
See my suggestion above to remove this. As a general rule of thumb, try to keep the number of names you introduce into the codebase as small as possible.

Just because something has a name doesn't mean that the name is more descriptive than the thing being named.

For example `T | null` is much more readable and descriptive than `Option<T>`. There's two reasons for this:
1. `T | null` will be familiar to every single TS/JS dev in the world, unlike `Option<T>` which could be implemented as `T | null` or `T | undefined` or `T | null | undefined`
2. A name's usefulness increases in proportion to how often the name is used in the codebase. Assuming that Option<T> was implemented as `T | null | undefined` and this pattern was used everywhere in the codebase, then that probably OK. Now consider the `TableDedOverrides` which is used only once (maybe twice?) in the entire codebase. People that look at that name will always have to look at the implementation.